src/he/conn.c

```diff
@@ -911,6 +945,34 @@ bool he_conn_is_password_set(const he_conn_t *conn) {
   return !he_internal_config_is_empty_string(conn->password);
 }
 
+he_return_code_t he_conn_set_auth_token(he_conn_t *conn, const uint8_t *token, size_t len) {
+  // Return if conn or token is null
+  if(!conn || !token) {
+    return HE_ERR_NULL_POINTER;
+  }
+  if(len == 0) {
+    return HE_ERR_EMPTY_STRING;
+  }
+  if(len >= HE_MAX_MTU) {
```

Comment:
Is `>=` because we need to save space for a null-terminating byte in the `auth_buffer`? Otherwise, can't this be just `>`? 

I don't see a mention of saving 1 byte in the header comment for `auth_buffer` at src/he/he_internal.h, so I'm assuming there isn't such a need.

Reply:
Yes, we do want to preserve 1 byte for the null-terminating byte, to make the server handling code a bit cleaner. The api document of this function actually mentioned: `* @param len The length of the token, it must be smaller than HE_MAX_MTU` but I'll try make this a bit clearer. :-)